api/layer/object.go

```diff
@@ -1488,3 +1418,103 @@ func (n *layer) searchEverythingForRemove(ctx context.Context, bkt *data.BucketI
 
 	return oids, nil
 }
+
+func (n *layer) searchAllVersionsInNeoFSWithCursor(ctx context.Context, bkt *data.BucketInfo, owner user.ID, objectName string, onlyUnversioned bool, cursor *string) ([]allVersionsSearchResult, string, error) {
```

Comment:
It'd be easier to handle if this would be located nearby `searchAllVersionsInNeoFS`. The patch would be smaller.